components/common/FormFields.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { ChangeEventHandler } from "react"
+import {
+    Button,
+    Checkbox,
+    Flex,
+    FormControl,
+    FormErrorMessage,
+    FormLabel,
+    Input,
+} from "@chakra-ui/react"
+import styled from "@emotion/styled"
+import ErrorAlert from "./ErrorAlert"
+
+type TaskFieldInputProps = {
+    label: string
+    placeholder: string
+    isRequired: boolean
+    isInvalid?: string
+    formErrorMessage?: string
+    onChange: ChangeEventHandler<HTMLInputElement>
+}
+
+export function FormInputField({
+    label,
+    placeholder,
+    isRequired,
+    isInvalid,
+    formErrorMessage,
+    onChange,
+}: TaskFieldInputProps): JSX.Element {
+    if (isRequired) {
+        return (
+            <FormControl isInvalid={!isInvalid} isRequired>
+                <FormLabel>{label}</FormLabel>
+                <Input placeholder={placeholder} onChange={onChange} />
+                {formErrorMessage && (
+                    <FormErrorMessage>{formErrorMessage}</FormErrorMessage>
+                )}
+            </FormControl>
+        )
+    }
+    return (
+        <FormControl>
+            <FormLabel>{label}</FormLabel>
+            <Input placeholder={placeholder} onChange={onChange} />
+            {formErrorMessage && (
+                <FormErrorMessage>{formErrorMessage}</FormErrorMessage>
+            )}
+        </FormControl>
+    )
+}
+
+type CheckBoxFieldProps = {
+    label: string
+    isChecked: boolean
+    onChange: ChangeEventHandler<HTMLInputElement>
+}
+
+export function CheckBoxField({
+    label,
+    isChecked,
+    onChange,
+}: CheckBoxFieldProps): JSX.Element {
+    return (
+        <FormControl py="0.5rem">
+            <Flex flexDirection={"row"} alignItems={"baseline"}>
+                <FormLabel>{label}</FormLabel>
+                <Checkbox isChecked={isChecked} onChange={onChange} />
+            </Flex>
+        </FormControl>
+    )
+}
+
+type FormButtonProps = {
+    onClick:
+        | ((() => void) | undefined)
+        | ((event: React.MouseEvent) => Promise<void | null>)
```

Comment:
This type looks wonky. Wouldn't something like

```typescript
onClick: (event: React.MouseEvent) => void
```

work?

Reply:
It will work for onSubmit, yes